testrunner-running/src/main/scala/edu/illinois/cs/statecapture/CustomAbstractReflectionConverter.java

```diff
@@ -0,0 +1,53 @@
+package edu.illinois.cs.statecapture;
+
+import com.thoughtworks.xstream.converters.ConversionException;
+import com.thoughtworks.xstream.converters.UnmarshallingContext;
+import com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter;
+import com.thoughtworks.xstream.converters.reflection.ReflectionProvider;
+import com.thoughtworks.xstream.io.HierarchicalStreamReader;
+import com.thoughtworks.xstream.mapper.Mapper;
+
+import java.lang.reflect.Field;
+
+public abstract class CustomAbstractReflectionConverter extends AbstractReflectionConverter {
+
+    public CustomAbstractReflectionConverter(final Mapper mapper, final ReflectionProvider reflectionProvider) {
+        super(mapper, reflectionProvider);
+    }
+
+    @Override
+    public Object doUnmarshal(final Object result, final HierarchicalStreamReader reader,
+                              final UnmarshallingContext context) {
+        if (!UnmarshalChain.isInitialized()) {
+            UnmarshalChain.initializeChain(System.getProperty("currentClassInXStream"), System.getProperty("currentFieldInXStream"));
+        }
+        super.doUnmarshal(result, reader, context);
+        return result;
+    }
+
+    @Override
+    protected Object unmarshallField(final UnmarshallingContext context, final Object result, final Class<?> type,
+                                     final Field field) {
+        // Assume properties define the root node and should be initialized as such if not yet
+        if (!UnmarshalChain.isInitialized()) {
+            UnmarshalChain.initializeChain(System.getProperty("currentClassInXStream"), System.getProperty("currentFieldInXStream"));
+            System.out.println("INITIALIZED TO " + System.getProperty("currentClassInXStream") + "::" + System.getProperty("currentFieldInXStream"));
+        }
+        UnmarshalChain.pushNode(UnmarshalChain.makeUnmarshalFieldNode(field.getDeclaringClass().getName(), field.getName()));
```

Comment:
Let's have a comment here about keeping track of current field structure being unmarshalled.